File: cpp_base_pkg/src/hw_status_publisher_node.cpp
#include "rclcpp/rclcpp.hpp"
#include "interfaces_pkg/msg/hardware_status.hpp"


class HardwareStatusPublisherNode : public rclcpp::Node // MODIFY NAME
{

private:
    rclcpp::Publisher<interfaces_pkg::msg::HardwareStatus>::SharedPtr pub_;
    rclcpp::TimerBase::SharedPtr timer_;

    void publishHardwareStatus()
    {
        auto msg = interfaces_pkg::msg::HardwareStatus();
        msg.temperature = 57;
        msg.are_motor_ready = false;
        msg.debug_message = "Motors are too hot!";
        pub_->publish(msg);

    }

public:
    HardwareStatusPublisherNode() : Node("hardware_status_publisher_node") // MODIFY NAME
    {
            pub_ = this->create_publisher<interfaces_pkg::msg::HardwareStatus>("hardware_status_node",10);
            
            timer_ = this-> create_wall_timer(
                        std::chrono::seconds(1),
                        std::bind(&HardwareStatusPublisherNode::publishHardwareStatus,this));
            RCLCPP_INFO(this->get_logger(), "Hardware status publisher has been started");
    }

};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<HardwareStatusPublisherNode>(); // MODIFY NAME
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: py_base_pkg/py_base_pkg/hw_status_publisher_node.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from interfaces_pkg.msg import HardwareStatus

class HardwareStatusPublisherNode(Node): # MODIFY NAME
    def __init__(self):
        super().__init__("hardware_status_publisher_node") # MODIFY NAME
        self.hw_status_publisher_ = self.create_publisher(HardwareStatus, "hardware_status", 10)
        self.timer_  =self.create_timer(1.0, self.publish_hw_status)
        self.get_logger().info("HW status publisher has been started")
    def publish_hw_status(self):
        msg = HardwareStatus()
        msg.temperature = 45
        msg.are_motor_ready = True
        msg.debug_message = "Nothing special here"
        self.hw_status_publisher_.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    node = HardwareStatusPublisherNode() # MODIFY NAME
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()
